tools/hardware/utils/memory.py

```diff
@@ -27,12 +27,12 @@ def visitor(node: WrappedNode):
 
 def merge_memory_regions(regions: Set[Region]) -> Set[Region]:
     ''' Check all region and merge adjacent ones '''
-    all_regions = [dict(idx=idx, region=region, right_adj=None, left_adj=None)
-                   for (idx, region) in enumerate(regions)]
+    all_regions = [dict(region=region, right_adj=None, left_adj=None)
+                   for region in regions]
 
     # Find all right contiguous regions
-    for dreg in all_regions:
-        for dnreg in all_regions[dreg['idx']+1:]:
+    for idx, dreg in enumerate(all_regions):
+        for dnreg in all_regions[idx+1:]:
```

Comment:
Are we guaranteed that this preserves the order in which the regions are visited? In the first line, we're making a new dict, and it doesn't seem to me like that will necessarily have the same order of elements in `enumerate` as the old `regions` dict.

It's not clear to me if that is semantically important, but if we're changing behaviour based on linter suggestions, I'd like to understand properly what the effects are first.

What was the linter complaining about in the original code?

Reply:
Just saw the commit message and I get it now, it's a list of `dicts`, not a `dict`, so the order is the same. Happy with that.